ext/CatalystStructuralIdentifiabilityExtension/structural_identifiability_extension.jl

```diff
@@ -113,7 +113,7 @@ function SI.assess_identifiability(rs::ReactionSystem, args...;
     funcs_to_check = make_ftc(funcs_to_check, conseqs, vars)
 
     # Computes identifiability and converts it to a easy to read form.
-    out = SI.assess_identifiability(osys, args...; measured_quantities,
+    out = SI.assess_identifiability(osys::ODESystem, args...; measured_quantities,
```

Comment:
This is the bit that solves it. What actually goes on is a royal mystery, but there is some discussion here: https://discourse.julialang.org/t/how-to-debug-an-error-in-type-inference/126122/8 and here: https://github.com/SciML/StructuralIdentifiability.jl/issues/360